Sources/SwiftParser/Expressions.swift

```diff
@@ -474,18 +474,51 @@ extension Parser {
         )
       )
 
+    case (.copyKeyword, let handle)?:
+      // `copy` is only contextually a keyword, if it's followed by an
+      // identifier or keyword on the same line. We do this to ensure that we do
+      // not break any copy functions defined by users. This is following with
+      // what we have done for the consume keyword.
+      let next = peek()
+
+      // We do this check early so it applies to identifier, dollarIdentifier,
+      // and the self check.
+      if next.isAtStartOfLine {
+        break
+      }
+
+      if next.rawTokenKind != .identifier,
+         next.rawTokenKind != .dollarIdentifier,
+         !(TokenSpec(.`self`) ~= next) {
+        break
+      }
```

Comment:
I’d prefer to use a `TokenSpec` here for all the cases and then just `switch` over `next`. It’s just more consistent with the rest of the code base and the other parsing primitives like `at(anyIn:)` or `canRecoverTo()`

```suggestion 
      switch self.peek() {
      case TokenSpec(.identifier, allowAtStartOfLine: false),
        TokenSpec(.dollarIdentifier, allowAtStartOfLine: false),
        TokenSpec(.self, allowAtStartOfLine: false):
        break
      default:
        break EXPR_PREFIX // And add an EXPR_PREFIX to the outer `switch`
      }
```